---
title: "General stats about the assembled contigs so far and R-S nucleotides"
output: github_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='images/',
                      echo=TRUE, warning=FALSE, message=FALSE)
```


## Nucleotide counts for 3 species

```{r load_data}
library(reshape)
library( dplyr )
library( ggplot2 )
library( tidyr )
library( data.table )
library(plyr)

#load stats R-S
allFiles <- list("/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/OutputKollectorRuns/PG29/RSstatsPG29.txt",
                  "/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/OutputKollectorRuns/WS77111/RSstatsWS77111.txt",
                 "/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/OutputKollectorRuns/Q903/RSstatsQ903.txt")
#load
l <- lapply( allFiles, function( fn ){
  d <- read.table( fn, header = F );
  d$fileName <- fn;
  d
  } );
#make df and refine
rs <- bind_rows( l );
rs$species = as.factor(sapply(strsplit(rs$fileName,"/"),"[[",11))
colnames(rs)[1:3] = c("contig","R","S")
rs$runKollector = as.factor(ifelse(grepl("extra",rs$contig)=="TRUE","V1extra","V1"))
```

## Stats 

```{r stats}
#-----------------------------------R
table(subset(rs, rs$R == 0)[,c("runKollector","species")])
#ratio between the 2 <- shown in summary
table(subset(rs, rs$R == 0)[,c("runKollector","species")])/table(rs[,c("runKollector","species")])
#------------------------------------------------------------
#ratio total and containing R only for species <-shown isn summary
table(subset(rs, rs$R == 0)[,c("species")])/table(rs[,c("species")])

#-----------------------------------S
table(subset(rs, rs$S == 0)[,c("runKollector","species")])
#ratio between the 2 <- shown in summary
table(subset(rs, rs$S == 0)[,c("runKollector","species")])/table(rs[,c("runKollector","species")])
#------------------------------------------------------------
#ratio total and containing R only for species <-shown isn summary
table(subset(rs, rs$S == 0)[,c("species")])/table(rs[,c("species")])

#see how many sequences have both
rs_both = subset(rs,rs$R > 0 & rs$S > 0 )
table(rs_both$species)
```


## Plot the distribution for R and S per sequence

```{r plotRS}
rs1_m = melt(rs[,c("contig","R","S")], id=c("contig"))
rs1_m = merge(rs1_m,rs[,c("contig","species")],by="contig")
ggplot(rs1_m ,aes(x=value, fill=variable)) + geom_histogram(alpha=.8,binwidth=1) + 
  facet_grid(species ~.) + 
  coord_cartesian(xlim = c(1, 4),ylim = c(1,200))

#plot colored by iteration
rs1_m$iter = as.factor(sapply(strsplit(rs1_m$contig,"="),"[[",2))
ggplot(subset(rs1_m,rs1_m$variable=="S" & rs1_m$value>0 ), aes(x=value, fill=iter)) + geom_density(alpha=.3)+
    facet_grid(species ~.)
ggplot(subset(rs1_m,rs1_m$variable=="R" & rs1_m$value>0 ), aes(x=value, fill=iter)) + geom_density(alpha=.3)+
    facet_grid(species ~.)
```

## Summary

- R (A or G) and S (G or C) are commonly assigned when the assembly does not recognize a certain pattern. It can mean both uncertainty but also gene families or differet alleles.
- In our case the percentage of sequences with such a nucleotide is very low (1-2%) but it need to be considered.
- The table shows the number of sequences with R or S:

| Nucl\species | PG29 | Q903 | WS77111 |
|:------------:|:----:|:----:|:-------:|
|       R      |  38  |  156 |   124   |
|       S      |  11  |  44  |    53   |
|     Total    |  49  |  200 |   177   |
|   Total cum. |  40  |  195 |   150   | 