---
title: "General stats about the assembled contigs so far and Ns"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='images/',
                      echo=TRUE, warning=FALSE, message=FALSE)
```

## Nucleotide counts for 3 species


```{r data_loading}
library( dplyr )
library( ggplot2 )
library( tidyr )
library( data.table )
library(plyr)

#list the files
allFiles <- list("/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/OutputKollectorRuns/PG29/NstatsPG29.txt",
                  "/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/OutputKollectorRuns/WS77111/NstatsWS77111.txt",
                 "/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/OutputKollectorRuns/Q903/NstatsQ903.txt")
#load
l <- lapply( allFiles, function( fn ){
  d <- read.table( fn, header = F );
  d$fileName <- fn;
  d
  } );
#make df
d <- bind_rows( l );
dim(d)
nams = c("contig","len","A","C","G","T","N","cpg","fileName")
colnames(d) = nams
#add category
d$species = as.factor(sapply(strsplit(d$fileName,"/"),"[[",11))
d$ratio = d$N/d$len
#exclude total
d1 = subset(d,d$contig != "total")
summary_total = subset(d,d$contig == "total")
#add a flag for extra runs
d1$runKollector = as.factor(ifelse(grepl("extra",d1$contig)=="TRUE","V1extra","V1"))

#---------------------------------------------------------------------
#check general summary
head(summary_total)
summary_total$N/summary_total$len
#---------------------------------------------------------------------

#chech how many samples have 0 N in the sequence
table(subset(d1, d1$N == 0)[,c("runKollector","species")])
#show the total contigs
table(d1[,c("runKollector","species")])
#------------------------------------------------------------
#ratio between the 2 <- shown in summary
table(subset(d1, d1$N == 0)[,c("runKollector","species")])/table(d1[,c("runKollector","species")])
#do the same but ignore Kollector runs
table(d1[,c("species")])
table(subset(d1, d1$N == 0)[,c("species")])
#------------------------------------------------------------
#ratio total and containing N only for species <-shown isn summary
table(subset(d1, d1$N == 0)[,c("species")])/table(d1[,c("species")]) 
```

## Plot the Ns in each contig for the 3 species 

```{r plotsN}
##length sequences
cdat <- ddply(d1, "species", summarise, rating.med=median(len))
head(cdat)
ggplot(d1, aes(x=len, fill=species)) + geom_density(alpha=.3) +
    geom_vline(data=cdat, aes(xintercept=rating.med,  colour=species),
               linetype="dashed", size=1)
###plot the generals stats for the Ns (all contigs)
ggplot(d1, aes(x=N)) + geom_histogram(fill="gray",colour="black",bins=100)
###plot the general stats for the Ns (exclude zeros)
NozeroN = subset(d1,d1$N!=0)
ggplot(NozeroN, aes(x=N)) + geom_histogram(fill="darkred",colour="black",bins=100) +
    coord_cartesian(xlim = c(1, 1500))
###plot general stats zoomed regions
ggplot(NozeroN, aes(x=N)) + geom_histogram(fill="darkblue",colour="black",bins=100) +
    coord_cartesian(xlim = c(1, 500))
#see peaks most high
order(table(NozeroN$N),decreasing=T)[1:10]
#summary
summary(NozeroN$N)
######################################################
#Ns ratios
#exclude ) Ns from the analysis
d2 = subset(d1, d1$N != 0)
cdat <- ddply(d2, "species", summarise, rating.med=median(ratio))
head(cdat)
ggplot(d2, aes(x=ratio)) +
    geom_histogram(binwidth=.001, colour="black", fill="white") +
    facet_grid(species ~.) +
    geom_vline(data=cdat, aes(xintercept=rating.med),
               linetype="dashed",colour="red", size=1)
#plot per Kollector and species separately
#--------------------------------------------
#WS77111
d2WS = subset(d2,d2$species == "WS77111")
ggplot(d2WS, aes(x=ratio, fill=runKollector)) + geom_density(alpha=.3)
d2PG = subset(d2,d2$species == "PG29")
ggplot(d2PG, aes(x=ratio, fill=runKollector)) + geom_density(alpha=.3)
d2Q903 = subset(d2,d2$species == "Q903")
ggplot(d2Q903, aes(x=ratio, fill=runKollector)) + geom_density(alpha=.3)

##and all together on a single figure
ggplot(d2, aes(x=ratio, fill=runKollector)) + geom_density(alpha=.3) +
    facet_grid(species ~.)
ddply(d2,species~runKollector,summarise,mean=mean(ratio),sd=sd(ratio),median=median(ratio), mad=mad(ratio))

## interesting: plot per V1 colored by iteration
d1$iter = as.factor(sapply(strsplit(d1$contig,"="),"[[",2))
#d12plot = subset(d1,d1$runKollector == "V1" & d1$ratio > 0 )
d12plot = subset(d1,d1$ratio > 0 )
#plot the 3 species
ggplot(d12plot, aes(x=ratio, fill=iter)) + geom_density(alpha=.3) +
    facet_grid(species ~.)
```

## Summary

- The total amount of Ns in the sequences is different for the 3 species. PG29 have the lower ratio (ration Ns/len seq), Q903 the highest

| species | ratio N/len |
|:-------:|:-----------:|
|   PG29  |    0.0222   |
| WS77111 |    0.0197   |
|   Q903  |    0.0173   |

- The number of sequences with no Ns at all is around 44.5, 48 and 44.8% respectively for PG29, WS77111 and Q903, see Table2. More detailed description in Table1.

**Table1**

| Run\species |    PG29    |     Q903    |  WS77111  |
|:-----------:|:----------:|:-----------:|:---------:|
|      V1     |  0.4492533 |  0.4799840  | 0.4495764 |
|   V1extra   |  0.3605578 |  0.4687500  | 0.3602941 |

**Table2**

|    PG29    |     Q903    |  WS77111  |
|:----------:|:-----------:|:---------:|
| 0.4454429  |  0.4799840  | 0.4448904 |

- The longest sequences are assembled in WS
- The size of N gaps is most frequently equal to 1N, than 200, 201, 50, 2, 400, 202, 51, 401 and 250. See peaks in the histograms.
- The median ratio Ns/len is higest in PG29, followed by WS77111. Q903 contains less Ns.
- The Ns/len ratio (after the exclusion of 0 Ns so no gaps) looks like a bimodal distribution. Not sure by what it depends. Maybe lower s parameter in Kollector since the second bump is observed mostly in KollectorV1 extra. Or maybe in for genes with a higher amount of introns which are assembled in the following iterations.
- **IMPORTANT: gaps in Abyss** - this is where we have higher frequency of some gaps. 1 -  a single lowercase “n” is used to indicate that two sequences overlap (due to paired-end evidence), but were not merged because the sequences did not perfectly match, 50 bp is the minimum gap length during the ABySS scaffolding stage, 200 bp is the minimum gap length during the ABySS long scaffolding stage. A peak in 400 may indicate 2 gaps of 200. Scaffolding is unabled by default in Abyss.