---
title: "Compare TPM"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='images/',
                      echo=TRUE, warning=FALSE, message=FALSE)
```

## Analysis gene expression - genome

### BUSCO genes - masked repeats

```{r loadData}
library(reshape)
library(ggplot2)

ws_quant = read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/RepeatsMaskedGenes/BUSCOs/WS77111/Q903_WS_quant/quant.sf",header=T)
ws_quantGF = read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/RepeatsMaskedGenes/GeneFamilies/WS77111/Q903_WS_quant/quant.sf", header = T)
q_quantGF = read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/RepeatsMaskedGenes/GeneFamilies/Q903/Q903_WS_quant/quant.sf",header=T)
q_quant = read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/RepeatsMaskedGenes/BUSCOs/Q903/Q903_WS_quant/quant.sf", header = T)

par(mfrow=c(1,2))
{plot(ws_quant$TPM,q_quant$TPM,xlim=c(0,10000),ylim=c(0,10000), xlab = "WS busco - TPM", ylab = "Q903 busco - TPM") 
abline(a=1, b=1)}
{plot(ws_quant$NumReads,q_quant$NumReads,xlim=c(0,10000),ylim=c(0,10000),xlab = "WS busco - num reads", ylab = "Q903 busco - num reads") 
abline(a=1, b=1)}

summary(ws_quant$NumReads)
summary(q_quant$NumReads)

summary(ws_quant$TPM)
summary(q_quant$TPM)

cor(ws_quant$TPM,q_quant$TPM)
cor(ws_quant$NumReads,q_quant$NumReads)

#-------------------------------------------------------------------------------
##Calculate Z score - WS77111
#POPULATION PARAMETER CALCULATIONS

counts = ws_quant$TPM
hist(log10(counts))
hist(log(counts))

logTrans <- function(vals) {
   vals_log = log(vals)
   vals_log[is.infinite(vals_log)]<- NA
   return(vals_log)
}

counts_log = logTrans(counts)

pop_sd <- sd(counts_log,na.rm = T)*sqrt((length(counts_log)-1)/(length(counts_log)))
pop_mean <- mean(counts_log, na.rm = T)

calc_zScore <- function(vector_vals, sd, mean) {
   Zvector_vals = (vector_vals - mean) / sd
   return(Zvector_vals)
}

ws_quantGF$z_vals = calc_zScore(logTrans(ws_quantGF$TPM), pop_sd, pop_mean)
sum(abs(ws_quantGF$z_vals) > 2, na.rm = T)
#-------------------------------------------------------------------------------
##Calculate Z score - Q903
#POPULATION PARAMETER CALCULATIONS

counts = q_quant$TPM
hist(log10(counts))
hist(log(counts))

counts_log = logTrans(counts)

pop_sd <- sd(counts_log,na.rm = T)*sqrt((length(counts_log)-1)/(length(counts_log)))
pop_mean <- mean(counts_log, na.rm = T)

q_quantGF$z_vals = calc_zScore(logTrans(q_quantGF$TPM), pop_sd, pop_mean)
sum(abs(q_quantGF$z_vals) > 2, na.rm = T)

#-----------------------------------------------------------------------
#plot values - Q903
q_quant$TPMlog = logTrans(q_quant$TPM)
q_quantmelt = melt(q_quant[,c("Name","TPMlog")])
q_quantmelt$variable = rep("BUSCO",nrow(q_quantmelt))
q_quantGF$TPMlog = logTrans(q_quantGF$TPM)
q_quantGFmelt = melt(q_quantGF[,c("Name","TPMlog")])
q_quantGFmelt$variable = rep("gene_family",nrow(q_quantGFmelt))

q_quantAll = rbind(q_quantGFmelt,q_quantmelt)

ggplot(q_quantAll, aes(x = value, fill = variable)) + 
  geom_density(alpha = 0.5) + 
  coord_cartesian(xlim = c(-10, 15))


ggplot(q_quantAll, aes(x = value, fill = variable)) + 
  geom_histogram(alpha = 0.5) +
  coord_cartesian(xlim = c(-10, 15))

#plot values - WS77111
ws_quant$TPMlog = logTrans(ws_quant$TPM)
ws_quantmelt = melt(ws_quant[,c("Name","TPMlog")])
ws_quantmelt$variable = rep("BUSCO",nrow(ws_quantmelt))
ws_quantGF$TPMlog = logTrans(ws_quantGF$TPM)
ws_quantGFmelt = melt(ws_quantGF[,c("Name","TPMlog")])
ws_quantGFmelt$variable = rep("gene_family",nrow(ws_quantGFmelt))

ws_quantAll = rbind(ws_quantGFmelt,ws_quantmelt)

ggplot(ws_quantAll, aes(x = value, fill = variable)) + 
  geom_density(alpha = 0.5) + 
  coord_cartesian(xlim = c(-10, 15))

ggplot(ws_quantAll, aes(x = value, fill = variable)) + 
  geom_histogram(alpha = 0.5,color="black") + 
  coord_cartesian(xlim = c(-10, 15))

table(ws_quantAll$variable)
```

### BUSCO genes - Transcript Sequences

```{r loadDataTrans}

ws_quantTrans <- read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/Transcripts/BUSCO/WS77111/Q903_WS_quant/quant.sf",header = T)
q_quantTrans <- read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/Transcripts/BUSCO/Q903/Q903_WS_quant/quant.sf", header = T)
ws_quantTransGF <- read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/Transcripts/GeneFamilies/WS77111/Q903_WS_quant/quant.sf",header = T)
q_quantTransGF <- read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/Transcripts/GeneFamilies/Q903/Q903_WS_quant/quant.sf", header = T)


par(mfrow=c(1,2))
{plot(ws_quantTrans$TPM,q_quantTrans$TPM,xlim=c(0,10000),ylim=c(0,10000), xlab = "WS busco - TPM", ylab = "Q903 busco - TPM") 
abline(a=1, b=1)}
{plot(ws_quantTrans$NumReads,q_quantTrans$NumReads,xlim=c(0,10000),ylim=c(0,10000),xlab = "WS busco - num reads", ylab = "Q903 busco - num reads") 
abline(a=1, b=1)}

summary(ws_quantTrans$NumReads)
summary(q_quantTrans$NumReads)

summary(ws_quantTrans$TPM)
summary(q_quantTrans$TPM)

cor(ws_quantTrans$TPM,q_quantTrans$TPM)
cor(ws_quantTrans$NumReads,q_quantTrans$NumReads)

#-------------------------------------------------------------------------------
##Calculate Z score - WS77111
#POPULATION PARAMETER CALCULATIONS

counts = ws_quantTrans$TPM
hist(log10(counts))
hist(log(counts))

logTrans <- function(vals) {
   vals_log = log(vals)
   vals_log[is.infinite(vals_log)]<- NA
   return(vals_log)
}

counts_log = logTrans(counts)

pop_sd <- sd(counts_log,na.rm = T)*sqrt((length(counts_log)-1)/(length(counts_log)))
pop_mean <- mean(counts_log, na.rm = T)

calc_zScore <- function(vector_vals, sd, mean) {
   Zvector_vals = (vector_vals - mean) / sd
   return(Zvector_vals)
}

ws_quantTransGF$z_vals = calc_zScore(logTrans(ws_quantTransGF$TPM), pop_sd, pop_mean)
sum(abs(ws_quantTransGF$z_vals) > 2, na.rm = T)
#-------------------------------------------------------------------------------
##Calculate Z score - Q903
#POPULATION PARAMETER CALCULATIONS

counts = q_quantTrans$TPM
hist(log10(counts))
hist(log(counts))

counts_log = logTrans(counts)

pop_sd <- sd(counts_log,na.rm = T)*sqrt((length(counts_log)-1)/(length(counts_log)))
pop_mean <- mean(counts_log, na.rm = T)

q_quantTransGF$z_vals = calc_zScore(logTrans(q_quantTransGF$TPM), pop_sd, pop_mean)
sum(abs(q_quantTransGF$z_vals) > 2, na.rm = T)

#-----------------------------------------------------------------------
#plot values - Q903
q_quantTrans$TPMlog = logTrans(q_quantTrans$TPM)
q_quantTransmelt = melt(q_quantTrans[,c("Name","TPMlog")])
q_quantTransmelt$variable = rep("BUSCO",nrow(q_quantTransmelt))
q_quantTransGF$TPMlog = logTrans(q_quantTransGF$TPM)
q_quantTransGFmelt = melt(q_quantTransGF[,c("Name","TPMlog")])
q_quantTransGFmelt$variable = rep("gene_family",nrow(q_quantTransGFmelt))

q_quantTransAll = rbind(q_quantTransGFmelt,q_quantTransmelt)

ggplot(q_quantTransAll, aes(x = value, fill = variable)) + 
  geom_density(alpha = 0.5) + 
  coord_cartesian(xlim = c(-10, 15))

ggplot(q_quantTransAll, aes(x = value, fill = variable)) + 
  geom_histogram(alpha = 0.5,color="black") + 
  coord_cartesian(xlim = c(-10, 15))

#plot values - WS77111
ws_quantTrans$TPMlog = logTrans(ws_quantTrans$TPM)
ws_quantTransmelt = melt(ws_quantTrans[,c("Name","TPMlog")])
ws_quantTransmelt$variable = rep("BUSCO",nrow(ws_quantTransmelt))
ws_quantTransGF$TPMlog = logTrans(ws_quantTransGF$TPM)
ws_quantTransGFmelt = melt(ws_quantTransGF[,c("Name","TPMlog")])
ws_quantTransGFmelt$variable = rep("gene_family",nrow(ws_quantTransGFmelt))

ws_quantTransAll = rbind(ws_quantTransGFmelt,ws_quantTransmelt)

ggplot(ws_quantTransAll, aes(x = value, fill = variable)) + 
  geom_density(alpha = 0.5) + 
  coord_cartesian(xlim = c(-10, 15))


ggplot(ws_quantTransAll, aes(x = value, fill = variable)) + 
  geom_histogram(alpha = 0.5,color="black") + 
  coord_cartesian(xlim = c(-10, 15))
```
```{r CompareGeneFamilies}

fcws_q = log(ws_quantGF$TPM) - log(q_quantGF$TPM)

sum(fcws_q > 0.58, na.rm = T)
```
```{r loadDataAverage}
ws_quantGFave = read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/RepeatsMaskedGenes/GeneFamilies/SalmonGeneRun/WS77111/Q903_WS_quant/quant.genes.sf", header = T)
q_quantGFave = read.delim("/projects/spruceup_scratch/dev/SprucePaper2018/GeneFamilies/Salmon/RepeatsMaskedGenes/GeneFamilies/SalmonGeneRun/Q903/Q903_WS_quant/quant.genes.sf",header=T)

par(mfrow=c(1,2))
{plot(ws_quantGFave$TPM,q_quantGFave$TPM,xlim=c(0,20000),ylim=c(0,20000), xlab = "WS gf - TPM", ylab = "Q903 gf - TPM") 
abline(a=1, b=1)}
{plot(ws_quantGFave$NumReads,q_quantGFave$NumReads,xlim=c(0,20000),ylim=c(0,20000),xlab = "WS gf - num reads", ylab = "Q903 gf - num reads") 
abline(a=1, b=1)}

#-----------------------------------------------------------------------
#plot values - Q903
q_quantGFave$TPMlog = logTrans(q_quantGFave$TPM)
q_quantmeltGFave = melt(q_quantGFave[,c("Name","NumReads")])
q_quantmeltGFave$variable = rep("gene_families_Q",nrow(q_quantmeltGFave))

q_quantAllGFave = rbind(q_quantmeltGFave,q_quantmelt)

#------------------------------------------------------------------------
#plot values - WS
ws_quantGFave$TPMlog = logTrans(ws_quantGFave$TPM)
ws_quantmeltGFave = melt(ws_quantGFave[,c("Name","NumReads")])
ws_quantmeltGFave$variable = rep("gene_families_WS",nrow(ws_quantmeltGFave))

q_ws_quantAllGFaveReads = rbind(ws_quantmeltGFave,q_quantmeltGFave)

ggplot(q_ws_quantAllGFaveReads, aes(x = value, fill = variable)) + 
  geom_density(alpha = 0.5) + 
  coord_cartesian(xlim = c(0, 200000))

ggplot(q_ws_quantAllGFaveReads, aes(x = value, fill = variable)) + 
  geom_histogram(alpha = 0.5) + 
  coord_cartesian(xlim = c(0, 500000))

#plot against BUSCOs
q_quantmeltGFaveTPM = melt(q_quantGFave[,c("Name","TPMlog")])
q_quantmeltGFave$variable = rep("gene_families",nrow(q_quantmeltGFaveTPM))

q_ws_quantAllGFaveTPM = rbind(q_quantmeltGFave,q_quantmelt)

ggplot(q_ws_quantAllGFaveTPM, aes(x = value, fill = variable)) + 
  geom_histogram(alpha = 0.5) + 
  coord_cartesian(xlim = c(-10,15))

```

