---
title: "Untitled"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='images/',
                      echo=TRUE, warning=FALSE, message=FALSE)
```

## Analysis TPS transcripts

```{r}
library( dplyr )
library( ggplot2 )
library( tidyr )
library( data.table )

dataPath="/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/CoverageKollectorWS"
allFiles <- list.files( path = dataPath, pattern = ".out", full.names = TRUE )

l <- lapply( allFiles, function( fn ){
  d <- read.table( fn, header = F );
  d$fileName <- fn;
  d
  } );

d <- bind_rows( l );
dim(d)
d$bin = sapply(strsplit(sapply(strsplit(d$fileName,"/"),"[[",11),"\\="),"[[",1)
d$iteration = gsub("iteration.", "", gsub(".out","",sapply(strsplit(sapply(strsplit(d$fileName,"/"),"[[",11),"\\="),"[[",2)))
colnames(d)[c(1:3)] = c("cov","trans","contig")
d <- d %>% filter(cov != 0)

#calculate the number of contigs per target
d1 = d %>% group_by( trans,iteration ) %>% summarize( freq = n(),cov_mean = mean(cov), cov_std = sd(cov) )
#replace NA in sd
d1$cov_std[which(is.na(d1$cov_std))] <- 0
head(d1)
```

## Load the TPS and select only those from the data set

```{r}
TPS = read.table("/projects/spruceup/interior_spruce/PG29/annotation/putative-terpene-related-targets.txt")
TPS$V1 = as.character(TPS$V1)
head(TPS,n=9)
TPS1 = d[which(d$trans%in% as.character(TPS$V1)),]
#number of temptative reconstructed TPS
length(unique(TPS1$trans))
#the ones that are NOT present in our reconstruction
TPS$V1[!(TPS$V1 %in% unique(TPS1$trans))]
#how many of those are considered as reconstructed
unique(subset(TPS1,TPS1$cov >= 0.9)$trans)
#maker-168427220-augustus-gene-0.5-mRNA-1 is reconstructed 2 times with cov 0.99
subset(TPS1,TPS1$cov >= 0.9)$trans
#get the counts
d1[which(d1$trans%in% as.character(TPS$V1)),] %>% print(n=40)
d2 = merge(TPS1[,c("trans","cov","iteration")],d1[which(d1$trans%in% as.character(TPS$V1)),],by=c("trans","iteration"))
d2 = distinct(d2,trans, cov, iteration , freq, cov_mean , cov_std )
```

## Plot the coverage for TPS

```{r}
d2$freq = as.factor(d2$freq)
ggplot(d2, aes(freq,cov)) +
    geom_boxplot(outlier.size=NA) +
    xlab("Number of reconstruceted genes per target") +
    ylab("Coverage") +
    theme(axis.title.x = element_text(face='bold',size=16,hjust=0.5),
          axis.title.y = element_text(face='bold',size=16,vjust=1),
          axis.text.x = element_text(face='bold',size=14,color='black'),
          axis.text.y = element_text(face='bold',size=14,color='black')) +  
    geom_hline(yintercept = 0.9,colour="red",linetype="dashed") + 
    geom_point(aes(colour = factor(trans)),size = 3) + 
    facet_wrap( ~ iteration ) + 
    scale_colour_discrete(name  ="TPS gene")
```

## Get the names of the 2 genes reconstructed with .99

```{r}
subset(d,d$trans=="maker-168427220-augustus-gene-0.5-mRNA-1")$contig
```

## Summary

* There are 8/9 TPS identified in our reconstructed gene set
* 3 out of those 8 are considered as successfully reconstructed, maker-168427220-augustus-gene-0.5-mRNA-1 is reconstructed with 2 genes
* 5/8 are reconstructed with > 1 gene

