---
title: "Paralogous analysis - WS77111"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='images/',
                      echo=TRUE, warning=FALSE, message=FALSE)
```

## See coverage trend per iteration

```{r}
library( dplyr )
library( ggplot2 )
library( tidyr )
library( data.table )

dataPath="/projects/btl/kgagalova/PHD_projects/SpruceUp/KollectorGeneReconstruction3species/RunKollector/data/CoverageKollectorWS"
allFiles <- list.files( path = dataPath, pattern = ".out", full.names = TRUE )

l <- lapply( allFiles, function( fn ){
  d <- read.table( fn, header = F );
  d$fileName <- fn;
  d
  } );

d <- bind_rows( l );
dim(d)
d$bin = sapply(strsplit(sapply(strsplit(d$fileName,"/"),"[[",11),"\\="),"[[",1)
d$iteration = gsub("iteration.", "", gsub(".out","",sapply(strsplit(sapply(strsplit(d$fileName,"/"),"[[",11),"\\="),"[[",2)))
colnames(d)[c(1:3)] = c("cov","trans","contig")
d <- d %>% filter(cov != 0)
```

## Show the number of assembled geness per transcript with any coverage

```{r boxplots_cov}
d1 = d %>% group_by( trans,iteration ) %>% summarize( freq = n(),cov_mean = mean(cov), cov_std = sd(cov) )
#replace NA in sd
d1$cov_std[which(is.na(d1$cov_std))] <- 0
head(d1)
```


plot the mean - sd in correspondence to the number of transcripts

```{r}
pd <- position_dodge(0.1)
d1$freq = as.factor(d1$freq)

d1_1=subset(d1,d1$iteration == 1)

#PLOTS

##plot only the first iteration
ggplot(d1_1, aes(freq,cov_mean,colour=cov_std)) +
    geom_boxplot(outlier.size=NA) +
    geom_jitter(aes(freq,cov_mean),
               position=position_jitter(width=0.1,height=0),
               alpha=0.6,
               size=3) +
    xlab("Number of reconstruceted genes per target") +
    ylab("mean coverage") +
    theme(axis.title.x = element_text(face='bold',size=16,hjust=0.5),
          axis.title.y = element_text(face='bold',size=16,vjust=1),
          axis.text.x = element_text(face='bold',size=14,color='black'),
          axis.text.y = element_text(face='bold',size=14,color='black')) + 
    geom_hline(yintercept = 0.9,colour="red",linetype="dashed") + 
    scale_colour_gradientn(colours = terrain.colors(10))


ggplot(d1, aes(freq,cov_mean,colour=cov_std)) +
    geom_boxplot(outlier.size=NA) +
    geom_jitter(aes(freq,cov_mean),
               position=position_jitter(width=0.1,height=0),
               alpha=0.6,
               size=3) +
    xlab("Number of reconstruceted genes per target") +
    ylab("mean coverage") +
    theme(axis.title.x = element_text(face='bold',size=16,hjust=0.5),
          axis.title.y = element_text(face='bold',size=16,vjust=1),
          axis.text.x = element_text(face='bold',size=14,color='black'),
          axis.text.y = element_text(face='bold',size=14,color='black')) + 
    facet_wrap( ~ iteration ) + 
    geom_hline(yintercept = 0.9,colour="red",linetype="dashed") + 
    scale_colour_gradientn(colours = terrain.colors(10))
```

## Summary statistics

```{r}

#show the number of elements per boxplot - x is frequency, y is iteration
table(d1$freq,d1$iteration)

apply(table(d1$freq,d1$iteration),2,sum) #check the number of transcripts with only one target

#total elements
sum(table(d1$freq,d1$iteration))

sum(subset(d1,d1$freq==1)$cov_mean >= 0.9)

sum(subset(d1,d1$freq==1)$cov_mean >= 0.95)

sum(subset(d1,d1$freq==1)$cov_mean == 0.99)

#check the ones that appear in multiple iterations
d1_1freq=subset(d1,d1$freq == 1)
d1_1freq = as.data.frame(table(d1_1freq$trans))
dim(d1_1freq)#unique targets <-------THIS IS THE IMPORTANT NUMBER

#genes reconstructed with different amount in different iterations
d1_allfreq = as.data.frame(table(d1$trans,d1$freq))
colnames(d1_allfreq) = c("trans","n_genes","Freq")
subset(d1,d1$trans=="augustus_masked-158701749-processed-gene-0.0-mRNA-1")

subset(d1,d1$trans=="augustus_masked-102623972-processed-gene-0.0-mRNA-1")

#show summary stats per sd
d1%>%group_by(iteration,freq)%>%summarise(Mean=mean(cov_std), Max=max(cov_std), Min=min(cov_std), Median=median(cov_std)) %>% print(n=25)
```

## Summary

**Only one gene reconstructed** 
* A good portion of transcripts has only one gene reconstructed after Kollector (independently of the iteration). The total is 16750/42574 - **39%** of all the genes generated during the runs.
* **Note that those nubers do not correspond to transcript number, some transcripts occur in multiple iterations. Unique transcripts are 13540 which correspond to 74.2% of the input Kollector targets** 
* The transcripts with only one gene reconstructed are shared between iterations in the following way (same transcripts may occur in different iterations): iteration 1: 10502 - iteration 2: 5119 - iteration 3: 552 - iteration 4: 260 - iteration 5: 317 
* 11098/16750 (**66.2%**) are successfully reconstructed based on the Kollector settings (success rate: coverage \>= 0.9). 10080/16750 (**60.17%**) have coverage \>= 0.95, 8410/16750 (**50.2%**) has coverage of **100%** (very high for reconstruction). 
* Since the gene is not reconstructed in some cases (coverage \< 0.9), there are some transcripts that occur in multiple iterations: 903 in 2 iterations, 523 in 3 iterations, 456 in 4 iterations and 570 in all the iterations.

**Variable genes reconstructed**
* There are some interesting cases where the same transcript is reconstructed with variable number of genes in different iterations.


